Mathlib/FieldTheory/RatFunc/AsPolynomial.lean

```diff
@@ -207,6 +210,96 @@ theorem valuation_of_mk (f : Polynomial K) {g : Polynomial K} (hg : g ≠ 0) :
       (Polynomial.idealX K).intValuation f / (Polynomial.idealX K).intValuation g := by
   simp only [RatFunc.mk_eq_mk' _ hg, valuation_of_mk']
 
+end HeightOneSpectrum
+
+section TrivialOnConstants
+
+variable {Γ : Type*}
+
+section LinearOrderedCommMonoidWithZero
+
+variable [LinearOrderedCommMonoidWithZero Γ]
+
+section Algebra
+
+variable (L : Type*) [Field L] [Algebra K L] {v : Valuation L Γ}
+
+lemma valuation_aeval_monomial_eq_valuation_pow
+    (w : L) (n : ℕ) {a : K}
+    (hv : ∀ a : K, a ≠ 0 → v (algebraMap K L a) = 1)
+    (ha : a ≠ 0) : v ((monomial n a).aeval w) = (v w) ^ n := by
+  simp [← C_mul_X_pow_eq_monomial, map_mul, map_pow, one_mul, RatFunc.algebraMap_X, hv a ha]
+
+theorem valuation_aeval_eq_valuation_X_pow_natDegree_of_one_lt_valuation_X [Nontrivial Γ]
+    [PosMulStrictMono Γ] (w : L) (hpos : 1 < v w) (hv : ∀ a : K, a ≠ 0 → v (algebraMap K L a) = 1)
+    {p : Polynomial K} (hp : p ≠ 0) : v (p.aeval w) = v w ^ p.natDegree := by
+  rw [← valuation_aeval_monomial_eq_valuation_pow _ _ _ _ hv ((leadingCoeff_ne_zero).mpr hp)]
+  nth_rw 1 [as_sum_range p, map_sum]
+  apply Valuation.map_sum_eq_of_lt _ (by simp) (by
+    replace hpos := Right.one_le_pow_of_le (hpos.le) (n := p.natDegree)
+    aesop)
+  intro i hi
+  simp only [Finset.mem_sdiff, Finset.mem_range, Finset.mem_singleton, Nat.lt_add_one_iff,
+    ← lt_iff_le_and_ne] at hi
+  simp_all only [ne_eq, RatFunc.algebraMap_eq_C, ← C_mul_X_pow_eq_monomial, map_mul,
```

Comment:
I am not on VSCode now, so this sequence of `simp` followed by `simp_all` seems strange, but there might be good reason (and ignore this comment in this case).

Reply:
I golfed this to avoid the `simp_all`.